src/server/controller.js

```diff
@@ -1,87 +1,189 @@
     }
   },
+  // Retrieve 10 most recent posts in Posts table
   posts: {
     get: function(req, res) {
-      // retrieve all posts in posts table
-      // filter to latest 10 results
-
-      // res.send('controller posts.get');
+      db.Post.findAndCountAll({
+        order: [['createdAt', 'DESC']],
+        limit: 10
+      })
+        .then(function(posts) {
+          res.json(posts);
+        });
     }
   },
+  // Retrieve all tags in Tags table
   tags: {
     get: function(req, res) {
-      // retrieve all tags from tags table
-
-      // res.send('controller tags.post');
+      db.Tag.findAll()
+        .then(function(tags) {
+          res.json(tags);
+        });
     }
   },
+  // Retrieve all categories in Categories table
   categories: {
     get: function(req, res) {
-      // retrieve all categories from categories table
-
-      // res.send('controller categories.get');
+      db.Category.findAll()
+        .then(function(categories) {
+          res.json(categories);
+        });
     }
   },
+  // Add a new post to database
   submit: {
     post: function(req, res) {
-      // add a newly created post to the posts table
-        // link, comment, tags, category
-        // initialize votes to 0
-
-      // res.send('controller submit.post');
+      db.Post.findOne({
+        where: {
+          problem_statement: req.body.problem,
+          resource: req.body.resource
+        }
+      })
+        .then(function(results) {
+          // Message if exact post has already been made
+          if (results !== null) {
+            console.log('Your message has already been posted');
+          // Create new post
+          } else {
+            db.Category.findOne({
+              where: { name: req.body.category }
+            })
+              .then(function(results) {
+                /**** TO DO: createdAt, updatedAt, CategoryId ****/
+                db.Post.create({
+                  problem_statement: req.body.problem,
+                  resource: req.body.resource,
+                  vote_count: 0,
+                  CategoryId: results.dataValues.id
+                })
+                  .then(function() {
+                    res.sendStatus(201);
+                  });
+              })
+          }
+        })
     }
   },
+  // Delete post from database
   delete: {
     post: function(req, res) {
-      // remove post from posts table
-
-      // res.send('controller delete.post');
+      db.Post.destroy({
+        where: { id: req.body.id },
+        limit: 1
+      })
+        .then(function(result) {
+          res.sendStatus(200);
+        });
     }
   },
+  // Increment vote count on post
   upvote: {
     post: function(req, res) {
-      // increment 'votes' on post in posts table
-
-      // res.send('controller upvote.post');
+      db.Post.findOne({
+        where: { id: req.body.id },
+      })
+        .then(function(result) {
+          result.increment('vote_count');
```

Comment:
It doesn't seem to need a call to save. The vote count persists in the database after increment or decrement.